chanman.go

```diff
@@ -37,19 +39,46 @@ func New(ctx context.Context, options *Options) *Chanman {
 func (cm *Chanman) Listen() {
 	defer cm.Quit()
 
+	jobs := make(chan interface{})
+
+	var wg sync.WaitGroup
+	for i := 0; i < cm.opts.Worker; i++ {
+		go worker(cm.opts.CallbackFn, jobs, &wg, cm.ctx)
+		wg.Add(1)
+	}
+
 Loop:
 	for {
 		select {
 		case <-cm.ctx.Done():
 			break Loop
 		case data, ok := <-cm.queueCh:
 			if ok {
-				cm.opts.CallbackFn(data)
+				jobs <- data
+			} else {
+				break Loop
+			}
+		}
+	}
+	wg.Wait()
+}
+
+// worker spawns simple runtime for concurrent worker pool
+func worker(callbackFn func(interface{}) error, jobs <-chan interface{}, wg *sync.WaitGroup, ctx context.Context) {
```

Comment:
ctx must be the first parameter and callbackFn is the last in best practices. So lets change it like this.